script.rb

```diff
@@ -0,0 +1,148 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield self[i]
+
+      i += 1
+    end
+    self
+  end
```

Comment:
- [ ] `my_each` is not working as the original `each` method, also is not passing all test cases.

### `each` is an Enumerator that has these characteristics: 

- it returns the array/range/hash itself if a block is given.
```
[1, 2, 3].each { |num| num }
# [1, 2, 3]
(1..3).each { |num| num * 2 }
# 1..3
````
- it applies the block for each element inside the array/range/hash
```
[1, 2, 3].each { |num| puts num }
# 1
# 2
# 3
# => [1, 2, 3]
```
[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerator.html#method-i-each)